### PR TITLE
Minor changes to Data Prepper index.md.

### PR DESCRIPTION
**Pull request by carolxob**
Signed-off-by: carolxob <carolxob@amazon.com>

### Description
This change addresses minor fixes, like typos or spacing issues on some of the main pages for Data Prepper documentation. Also addresses Data Prepper navigation/ToC.

### Issues Resolved
There are currently no issues associated with this PR.

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by carolxob** [comment]
Proposed re-ordering of overall ToC from @dlvenable :

Data Prepper
- Getting started
- Pipelines
- log analytics
- Core APIs
- Log4j configuration
- Configure Logstash for Data Prepper
- Migrating from Open Distro Data Prepper

**Review by alicejw-aws — APPROVED** [review]
Looks good!

**Review by cwillum — APPROVED** [review]
LGTM

**Comment by dlvenable** [comment]
I put together #2453 to propose a possible ToC structure. I'd be happy to have a discussion on the approach in that issue.

File: _data-prepper/getting-started.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ You will configure two files:
 * `data-prepper-config.yaml`
 * `pipelines.yaml`
 
-Depending on your use case, we have a few different guides to configuring Data Prepper.
+Depending on your use case, we have a few different guides to configure Data Prepper.
```
[the remark this answers is not in the document]
Comment (natebower) [reply _data-prepper/getting-started.md:43]: The original was correct here (guide to doing something).

File: _data-prepper/index.md [hunk 2]
```diff
@@ -12,18 +12,18 @@ redirect_from:
 
 # Data Prepper
 
-Data Prepper is a server side data collector capable of filtering, enriching, transforming, normalizing and aggregating data for downstream analytics and visualization.
+Data Prepper is a server-side data collector capable of filtering, enriching, transforming, normalizing and aggregating data for downstream analytics and visualization.
```
Comment (natebower) [inline _data-prepper/index.md:15]:
```suggestion
Data Prepper is a server-side data collector capable of filtering, enriching, transforming, normalizing, and aggregating data for downstream analytics and visualization.
```

**Review by natebower — COMMENTED** [review]
@carolxob Just a few minor changes. Thanks!

**Comment by hdhalter** [comment]
HI @carolxob - why are you waiting to merge this? Isn't this going in the 2.5 build? Thx.

**Comment by carolxob** [comment]
@hdhalter Only applied this label in case we needed to wait to merge. It is otherwise ready to go.

**Pull request merged**